src/annotation_operators.ts

```diff
@@ -238,56 +237,161 @@ function get_distance_from_point_to_line(point_annotation: ULabelAnnotation, lin
 }
 
 /**
- * Assigns each point annotation a distance from each diffrent class of row. Will also update the distance from any row.
+ * Wrapper for assign_closest_line_to_single_point()
  * 
  * @param point_annotations Set of point annotations
  * @param line_annotations Set of line annotations
  * @param offset Offset of a particular annotation in the set. Used when an annotation is being moved by the user
  */
-export function assign_distance_from_line(
+export function assign_closest_line_to_each_point(
     point_annotations: ULabelAnnotation[],
     line_annotations: ULabelAnnotation[],
-    offset: Offset = null) {
+    offset: Offset = null
+) {
     // Loop through every point and assign it a distance from line
     point_annotations.forEach(current_point => {
+        assign_closest_line_to_single_point(current_point, line_annotations, offset)
+    })
+}
 
-        // Create a DistanceFrom object to be assigned to this point
-        let distance_from: AnnotationClassDistanceData = {"single": undefined}
-
-        // Calculate the distance from each line and populate the distance_from accordingly
-        line_annotations.forEach(current_line => {
 
-            const line_class_id = get_annotation_class_id(current_line)
-            
-            const distance = get_distance_from_point_to_line(current_point, current_line, offset)
+/**
+ * Assigns a single point annotation a distance to the closest polyline
+ * from each diffrent class of polyline, as well as the distance from
+ * the closest line, regardless of class.
+ * 
+ * @param point_annotation single point annotation
+ * @param line_annotations set of line annotations
+ * @param offset Offset of a particular annotation in the set. Used when an annotation is being moved by the user
+ */
+export function assign_closest_line_to_single_point(
+    point_annotation: ULabelAnnotation,
+    line_annotations: ULabelAnnotation[],
+    offset: Offset = null
+) {
+    // Create a new distance_from object for the point annotation
+    let distance_from: DistanceFromPolylineClasses = {closest_row: {distance: Infinity}}
```

Comment:
yeah this was a fix to prevent errors when there are no lines. as far as i can tell there should be no issues, bc the distance is always just used for comparisons